generator/new_service.go

```diff
@@ -53,3 +60,19 @@ func (g *NewService) Generate() error {
 	)
 	return g.fs.WriteFile(g.filePath, g.srcFile.GoString(), false)
 }
+
+func (g *NewService) genModule() error {
+	exist, _ := g.fs.Exists(g.name + "/go.mod")
```

Comment:
Maybe we want to add the dependencies manually, at least the once we depend on after running `kit g s`

Reply:
Hi @kujtimiihoxha :）
Why we need to maintain multiple services under the same package？
I think it has two scenes：
The one scene is that we just maintain multiple services and convenient to manage the project. There have no relation between services and services are started by there own main.go file ({service}/cmd/main.go). 

The other scene is that we want to construct a monolithic project that contains multiple services under the same package. There have some dependency between services and the monolithic project is started only by one main.go file which needs to be create manually at the service's parent level. In addition,  we should also delete the go.mod file at the service level and create a go.mod file manually at the parent level. 

About the one scene, I think it is of small significance and rarely used in this way. We only need to cover most of the usage scenarios. There's no need to support.

About the other scene, I think it is significant. But I do not recommend realizing the function at the kit tool. The responsibility of kit tool is just construct our services.

Reply:
I think you are right.

One other thing that came to my mind now is that with how it is implemented now we are not maintaining backwards compatibility with already existing projects that use kit.

Is there a way to only use modules when the user wants to do that ? 

Basically only use mod if `--module` is provided OR the `go.mod` file is detected in the service folder.

Reply:
About compatible with the already existing projects，we may have two ways, I think.

The one is that we should use tag vesion to manage our feature, such as checkout two new branches named release/v0.1.0 and  release/v0.2.0 from the master and tag the release/v0.1.0 branch code v0.1.0 which should run kit commands under the GOPATH, and then switch to release/v0.2.0 branch, merge code from feature/mod_support branch, tag v0.2.0 under the release/v0.2.0 branch. If the user want to use the GOPATH ,he can use the v0.1.0 version; if the user want to use the module feature, he can use the v0.2.0 version.

The other is that continue to support GOPATH in our new version, if user run `kit n s {service}` without module flag, it will use the old way to new and generate service and client library. 

I think the first way is better than the second.  I want to quote some word from https://blog.golang.org/modules2019.

> Our aim is for Go 1.13, scheduled for August 2019, to enable module mode by default (that is, to change the default from auto to on) and deprecate GOPATH mode. In order to do that, we’ve been working on better tooling support along with better support for the open-source module ecosystem.

So, I think there is no need to compatible with GOPATH in our new version. We should guide the user to use the new module feature. If the user have to use the GOPATH, he could switch to the old version v0.1.0.

What do you think about this? :)